activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/event/impl/ToTimerExecutionFailureConverter.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.runtime.api.event.impl;
+
+import java.util.Optional;
+
+import org.activiti.api.process.model.builders.ProcessPayloadBuilder;
+import org.activiti.api.process.model.events.BPMNTimerExecutionFailureEvent;
+import org.activiti.api.process.model.payloads.TimerPayload;
+import org.activiti.api.runtime.event.impl.BPMNTimerExecutionFailureEventImpl;
+import org.activiti.api.runtime.model.impl.BPMNTimerImpl;
+import org.activiti.engine.delegate.event.ActivitiEntityEvent;
+import org.activiti.engine.impl.persistence.entity.JobEntityImpl;
+
+public class ToTimerExecutionFailureConverter implements EventConverter<BPMNTimerExecutionFailureEvent, ActivitiEntityEvent> {
+
+    public ToTimerExecutionFailureConverter() {
+    }
+
+    @Override
+    public Optional<BPMNTimerExecutionFailureEvent> from(ActivitiEntityEvent internalEvent) {
+        
+        JobEntityImpl jobEntity = (JobEntityImpl)internalEvent.getEntity();
+        
+        BPMNTimerImpl timer = new BPMNTimerImpl(jobEntity.getId());
```

Comment:
if a generic design `Optional<T> from(ActivitiEntityEvent internalEvent, Class<T> type) is not applicable,
maybe we can externalise to an external utility method the code block from line 39-58 that can be reused by the other converter classes too

Reply:
+1

Reply:
Refactoring made: https://github.com/Activiti/Activiti/pull/2814/commits/01fc43c359a119440d4b58c2dd78c6a64b52288c